src/bun.js/bindings/Os.cpp

```diff
@@ -0,0 +1,149 @@
+
+#include "root.h"
+
+#include "BunClientData.h"
+
+#include "JavaScriptCore/JSFunction.h"
+#include "JavaScriptCore/JSMicrotask.h"
+#include "JavaScriptCore/ObjectConstructor.h"
+
+#pragma mark - Node.js Os
+
+namespace Zig {
+
+static JSC::JSObject* createOs(JSC::JSGlobalObject* globalThis, bool isWindows);
+
+using JSGlobalObject = JSC::JSGlobalObject;
+using Exception = JSC::Exception;
+using JSValue = JSC::JSValue;
+using JSString = JSC::JSString;
+using JSModuleLoader = JSC::JSModuleLoader;
+using JSModuleRecord = JSC::JSModuleRecord;
+using Identifier = JSC::Identifier;
+using SourceOrigin = JSC::SourceOrigin;
+using JSObject = JSC::JSObject;
+using JSNonFinalObject = JSC::JSNonFinalObject;
+namespace JSCastingHelpers = JSC::JSCastingHelpers;
+
+using namespace JSC;
+
+// clang-format off
+#define DEFINE_CALLBACK_FUNCTION_BODY(ZigFunction) JSC::VM& vm = globalObject->vm(); \
```

Comment:
Instead of doing this, in the Zig code we can do

`JSC.NewFunction()` and in Zig the function will look like

```zig
fn myFunction(globalObject: *JSC.JSGlobalObject, callframe: *JSC.Callframe) callconv(.C) JSValue {

}
```

Then you won't have to write these C++ bindings and it will be a little faster at runtime.